### ISSUE TITLE
I need help

### ISSUE THREAD

**Issue by Artugr18** [iRedScarf/onedrive-index#10]
I am doing everything correctly but I still have a problem. Is there any way to contact you, e.g. via discord?

**Comment by Artugr18** [comment]
https://i.imgur.com/NP36J8S.png

I have something like this and now what to do

**Comment by Artugr18** [comment]
sorry missclick

**Comment by iRedScarf** [comment]
Error code 504 typically means "Gateway Timeout". It is an HTTP status code indicating that a server, while acting as a gateway or proxy, did not receive a timely response from another server or some auxiliary servers. In simpler terms, a 504 error occurs when your server tries to load a web page or resource but another server takes too long to respond.

Here are some common reasons that can lead to a 504 error:

1. **Server Overload:** The target server is too busy handling other requests to respond in time.
2. **Network Issues:** There are problems with the network connection between the servers, causing requests or responses to get lost.
3. **Incorrect Server Configuration:** Misconfiguration of the gateway or proxy server can lead to timeouts in processing requests.

Solutions to a 504 error may include checking server loads, inspecting network connections, and reviewing server and proxy configurations. Since this type of error often involves communication between servers, there's often limited action that can be taken from the user end.

Since I don't know the specific process of how you encountered the 504 error, it's difficult to analyze the exact cause of this issue. If you could provide more information, such as the platform on which you are deploying or whether you have correctly configured all the environment variables as per the documentation, it would be helpful.

**Comment by Artugr18** [comment]
I wanted to set up an index using microsoft onedrive as a personal account. I did every step in the tutorial and tried several times to generate files as a new repository via vercel. I also changed the value of BASE_DIRECTORY several times. Currently nothing has helped. Below are pictures of what I set.

https://imgur.com/a/GKOtEe8

**Comment by iRedScarf** [comment]
I'm deeply sorry for the delay in responding and providing assistance. 

It appears that you encountered a 504 error. I carefully reviewed the screenshots of your settings, and everything seems to be correct. I'm unable to pinpoint the exact source of the issue. 

However, I noticed an error message in the browser developer tools in your first screenshot: `Uncaught SyntaxError: expected expression, got '<'.` This error message is worth investigating. It's possible that there may be a syntax error in the settings files you modified, leading to the unsuccessful deployment of the project.

**Comment by Miclicks** [comment]
i have the same error

**Comment by Miclicks** [comment]
i think i could help you. I had the same error, i forgot last important steps:
After the initial successful deployment, the deployed page will return a 404 error because we still need to connect to the Redis database.
REDIS_URL:If you are encountering Redis database for the first time, I strongly recommend using Upstash, which is free and deeply integrated with Vercel. For details, refer to [Vercel Integration](https://docs.upstash.com/redis/howto/vercelintegration). Follow the instructions to set it up in Vercel's [Upstash Integration](https://vercel.com/integrations/upstash)(simply create a new database in the Redis of Upstash, then create a new integration in Vercel Integrations, and associate the just deployed OneDrive-Index project with the Redis database), it will automatically fill in the environment variables after project deployment.

After REDIS_URL is successfully set, redeploy the project again.

After successful deployment, when you visit your onedrive-vercel-index page for the first time, it will guide you to perform OAuth authentication (quite simple). For details, please refer to the [Instructions](https://ovi.swo.moe/zh/docs/getting-started#authentication) written by the original author.

Also remember to make obfuscated client secret here: https://ovi.swo.moe/docs/advanced#modify-api-permissions
Here you have to paste your client secret and create obfuscated client secret. REMEMBER to set in vercel  variables CLIENT_SECRET to your obfuscated client secret (not just client secret). Hope that helps!
